[docs] update eas.json page - managed projects support (#11192)
expo/expo: docs/pages/build/eas-json.md

  "credentialsSource": "local" | "remote" | "auto", // default: "auto"
  "withoutCredentials": boolean, // default: false
  "gradleCommand": string, // default: ":app:bundleRelease"
  "artifactPath": string, // default: "android/app/build/outputs/**/*.{apk,aab}"
  "releaseChannel": string, // default: "default"
  "distribution": "store" | "internal" // default: "store"
}
```

- `"workflow": "generic"` indicates that your project is a generic one.

> - `credentialsSource` defines the source of credentials for this build profile. If you want to take advantage of your own `credentials.json` file, set this to `local` ([learn more on this here](advanced-credentials-configuration.md)). If you want to use the credentials Expo already has stored for you, choose `remote`. If you're not sure what to do but you probably won't be running builds from a CI, choose `auto` (this is the default option).

Author: ```suggestion
- `credentialsSource` defines the source of credentials for this build profile. If you want to provide your own `credentials.json` file, set this to `local` ([learn more on this here](advanced-credentials-configuration.md)). If you want to use the credentials Expo already has stored for you, choose `remote`. If you're not sure what to do, choose `auto` (this is the default option).
```

- `withoutCredentials` when set to `true`, EAS CLI won't require you to configure credentials when building the app using this profile. It comes in handy when you want to build debug binaries and the debug keystore is checked in to the repository. The default is `false`.
- `gradleCommand` defines the Gradle task to be run on Expo servers to build your project. You can set it to any valid Gradle task, e.g. `:app:assembleDebug` to build a debug binary. The default Gradle command is `:app:bundleRelease`.
- `artifactPath` is the path (or pattern) where EAS Build is going to look for the build artifacts. EAS Build uses the `fast-glob` npm package for pattern matching, [see their README to learn more about the syntax you can use](https://github.com/mrmlnc/fast-glob#pattern-syntax). The default value is `android/app/build/outputs/**/*.{apk,aab}`.

> - `releaseChannel` is the release channel for the `expo-updates` package (if it's installed). [Learn more on this here](../distribution/release-channels.md). If you do not specify a channel, your binary will pull releases from the `default` channel.

Reviewer: ```suggestion
- `releaseChannel` is the release channel for the `expo-updates` package ([Learn more about this](../distribution/release-channels.md)). If you do not specify a channel, your binary will pull releases from the `default` channel. If you do not use `expo-updates` in your project then this property will have no effect.
```

> - `distribution` is the flow of distributing your app. If you choose `internal` you'll be able to share your build's URL with anyone and they will be able to install it straight from the Expo website. When using `internal`, make sure the `gradleCommand` produces an APK file (e.g. `:app:assembleRelease`). Otherwise, the sharable URL will be useless. The default is `store` which means your build's URL won't be sharable. [Learn more on Internal Distribution](internal-distribution.md).

Reviewer: ```suggestion
- `distribution` is the flow of distributing your app. If you choose `internal` you'll be able to share your build URLs with anyone, and they will be able to install the builds to their devices straight from the Expo website. When using `internal`, make sure the `gradleCommand` produces an APK file (e.g. `:app:assembleRelease`). Otherwise, the sharable URL will be useless. The default is `store` which means your build's URL won't be sharable. [Learn more on Internal Distribution](internal-distribution.md).
```

### Managed Project

The schema of a build profile for a managed Android project looks like this:

```json
{
  "workflow": "managed",
  "credentialsSource": "local" | "remote" | "auto", // default: "auto"
  "buildType": "app-bundle" | "apk", // default: "app-bundle"
  "releaseChannel": string, // default: "default"
  "distribution": "store" | "internal" // default: "store"
}
```

- `"workflow": "managed"` indicates that your project is a managed one.

> - `credentialsSource` defines the source of credentials for this build profile. If you want to take advantage of your own `credentials.json` file, set this to `local` ([learn more on this here](advanced-credentials-configuration.md)). If you want to use the credentials Expo already has stored for you, choose `remote`. If you're not sure what to do but you probably won't be running builds from a CI, choose `auto` (this is the default option).

Reviewer: ```suggestion
- `credentialsSource` defines the source of credentials for this build profile. If you want to provide your own `credentials.json` file, set this to `local` ([learn more on this here](advanced-credentials-configuration.md)). If you want to use the credentials Expo already has stored for you, choose `remote`. If you're not sure what to do, choose `auto` (this is the default option).
```

probably best to just leave this bit about CI out for now, people can figure it out when they look into configuring for CI

- `buildType` when set to `app-bundle` produces an AAB archive but when set to `apk` produces an APK instead.

> - `releaseChannel` is the release channel for the `expo-updates` package (if it's installed). [Learn more on this here](../distribution/release-channels.md). If you do not specify a channel, your binary will pull releases from the `default` channel.

Reviewer: ```suggestion
- `releaseChannel` is the release channel for the `expo-updates` package ([Learn more about this](../distribution/release-channels.md)). If you do not specify a channel, your binary will pull releases from the `default` channel. If you do not use `expo-updates` in your project then this property will have no effect.
```

> - `distribution` is the flow of distributing your app. If you choose `internal` you'll be able to share your build's URL with anyone and they will be able to install it straight from the Expo website. When `distribution` is `internal`, `buildType` needs to be set to `apk`. The default is `store` which means your build's URL won't be sharable. [Learn more on Internal Distribution here](internal-distribution.md).

Reviewer: ```suggestion
- `distribution` is the flow of distributing your app. If you choose `internal` you'll be able to share your build URLs with anyone, and they will be able to install the builds to their devices straight from the Expo website. When `distribution` is `internal`, `buildType` needs to be set to `apk`. The default is `store` which means your build's URL won't be sharable. [Learn more on Internal Distribution here](internal-distribution.md).
```

  "workflow": "generic",
  "credentialsSource": "local" | "remote" | "auto", // default: "auto"
  "scheme": string,
  "artifactPath": string, // default: "ios/build/App.ipa"
  "releaseChannel": string, // default: "default"
  "distribution": "store" | "internal" // default: "store"
}
```

- `"workflow": "generic"` indicates that your project is a generic one.

> - `credentialsSource` defines the source of credentials for this build profile. If you want to take advantage of your own `credentials.json` file, set this to `local` ([learn more on this here](advanced-credentials-configuration.md)). If you want to use the credentials Expo already has stored for you, choose `remote`. If you're not sure what to do but you probably won't be running builds from a CI, choose `auto` (this is the default option).

Author: ```suggestion
- `credentialsSource` defines the source of credentials for this build profile. If you want to provide your own `credentials.json` file, set this to `local` ([learn more on this here](advanced-credentials-configuration.md)). If you want to use the credentials Expo already has stored for you, choose `remote`. If you're not sure what to do, choose `auto` (this is the default option).
```

- `scheme` defines the Xcode project's scheme to build. You should set it if your project has multiple schemes. Please note that if the project has only one scheme, it will automatically be detected. However, if multiple schemes exist and this value is **not** set, EAS CLI will prompt you to select one of them.
- `artifactPath` is the path (or pattern) where EAS Build is going to look for the build artifacts. EAS Build uses the `fast-glob` npm package for pattern matching, [see their README to learn more about the syntax you can use](https://github.com/mrmlnc/fast-glob#pattern-syntax). You should modify that path only if you are using a custom `Gymfile`. The default is `ios/build/App.ipa`.

> - `releaseChannel` is the release channel for the `expo-updates` package (if it's installed). [Learn more on this here](../distribution/release-channels.md). If you do not specify a channel, your binary will pull releases from the `default` channel.

Reviewer: ```suggestion
- `releaseChannel` is the release channel for the `expo-updates` package ([Learn more about this](../distribution/release-channels.md)). If you do not specify a channel, your binary will pull releases from the `default` channel. If you do not use `expo-updates` in your project then this property will have no effect.
```

> - `distribution` is the flow of distributing your app. If you choose `internal` you'll be able to share your build's URL with anyone and they will be able to install it straight from the Expo website. The default is `store` which means your build's URL won't be sharable. [Learn more on Internal Distribution here](internal-distribution.md).

Reviewer: ```suggestion
- `distribution` is the flow of distributing your app. If you choose `internal` you'll be able to share your build URLs with anyone, and they will be able to install the builds to their devices straight from the Expo website. The default is `store` which means your build's URL won't be sharable. [Learn more on Internal Distribution here](internal-distribution.md).
```

### Managed Project

The schema of a build profile for a managed iOS project looks like this:

```json
{
  "workflow": "managed",
  "credentialsSource": "local" | "remote" | "auto", // default: "auto"
  "releaseChannel": string, // default: "default"
  "distribution": "store" | "internal" // default: "store"
}
```

- `"workflow": "managed"` indicates that your project is a managed one.

> - `credentialsSource` defines the source of credentials for this build profile. If you want to take advantage of your own `credentials.json` file, set this to `local` ([learn more on this here](advanced-credentials-configuration.md)). If you want to use the credentials Expo already has stored for you, choose `remote`. If you're not sure what to do but you probably won't be running builds from a CI, choose `auto` (this is the default option).

Author: ```suggestion
- `credentialsSource` defines the source of credentials for this build profile. If you want to provide your own `credentials.json` file, set this to `local` ([learn more on this here](advanced-credentials-configuration.md)). If you want to use the credentials Expo already has stored for you, choose `remote`. If you're not sure what to do, choose `auto` (this is the default option).
```

> - `releaseChannel` is the release channel for the `expo-updates` package (if it's installed). [Learn more on this here](../distribution/release-channels.md). If you do not specify a channel, your binary will pull releases from the `default` channel.

Author: ```suggestion
- `releaseChannel` is the release channel for the `expo-updates` package ([Learn more about this](../distribution/release-channels.md)). If you do not specify a channel, your binary will pull releases from the `default` channel. If you do not use `expo-updates` in your project then this property will have no effect.
```

> - `distribution` is the flow of distributing your app. If you choose `internal` you'll be able to share your build's URL with anyone and they will be able to install it straight from the Expo website. The default is `store` which means your build's URL won't be sharable. [Learn more on Internal Distribution here](internal-distribution.md).

Reviewer: ```suggestion
- `distribution` is the flow of distributing your app. If you choose `internal` you'll be able to share your build URLs with anyone, and they will be able to install the builds to their devices straight from the Expo website. The default is `store` which means your build's URL won't be sharable. [Learn more on Internal Distribution here](internal-distribution.md).
```
